app/js/arethusa.morph/services/morph.js

```diff
@@ -840,51 +678,188 @@ angular.module('arethusa.morph').service('morph', [
+      // special empty attributes allows to say specifically that something
+      // should be left unannotated/unknown. Useful for elliptic nodes etc.
+      addSpecialEmptyAttributes();
+
+      if (self.conf.lexicalInventory) {
+        inventory = configurator.getRetriever(self.conf.lexicalInventory.retriever);
+      }
+
+      colorMap = undefined;
+      searchIndex = {};
+
+      function addSpecialEmptyAttributes() {
+        angular.forEach(self.attributes, addSpecialEmptyAttribute);
+
+        function addSpecialEmptyAttribute(attrObj, name) {
+          attrObj.values['---'] = emptyAttribute;
+        }
+      }
+      function propagateMappings(retrievers) {
+        angular.forEach(retrievers, propagateMapping);
+        function propagateMapping(retriever, name) {
+          retriever.mapping = mappingFor(name);
+          function mappingFor(name) {
+            // this exists so that mapping instances can refer to each
+            // other through providing a string instead of an mappings
+            // object.
+            var mappings = self.mappings[name];
+            while (angular.isString(mappings)) {
+              mappings = self.mappings[name];
+            }
+            return mappings || {};
+          }
+        }
+      }
+    }
 
     this.init = function () {
+      // When a user is moving fast between chunks, a lot of outstanding
+      // requests can build up in the retrievers. As they are all asynchronous
+      // their callbacks fire when we have already moved away from the chunk which
+      // started the calls.
+      // This can lead to quite a bit of confusion and is generally not a very
+      // good solution.
+      // We therefore use the new abort() API of Resource to cancel all requests
+      // we don't need anymore. All morph retrievers need to provide an abort()
+      // function now (usually just a delegator to Resource.abort).
+      //
+      // On init, we check if morphRetrievers were already defined and if they
+      // are we abort all outstanding requests.
+      function abortOutstandingRequests() {
+        function abortRetriever(retriever) {
+          var fn = retriever.abort;
+          if (angular.isFunction(fn)) fn();
+        }
+
+        if (morphRetrievers) {
+          angular.forEach(morphRetrievers, abortRetriever);
+        }
+      }
       abortOutstandingRequests();
       configure();
+      function createEmptyPostag() {
+        return arethusaUtil.map(self.postagSchema, function (el) {
+          return '-';
+        }).join('');
+      }
       self.emptyPostag = createEmptyPostag();
       self.analyses = seedAnalyses();
       loadInitalAnalyses();
       loadSearchIndex();
       plugins.declareReady(self);
+
+      function seedAnalyses() {
+        return arethusaUtil.inject({}, state.tokens, function (obj, id, token) {
+          obj[id] = new Forms(token.string);
+        });
+      }
+      function loadInitalAnalyses() {
```

Comment:
@fbaumgardt we do have a unit test for this which will break now, but I think you could argue that the test should actually be on loadToken not loadInitialAnalyses anyway